packages/calcite-components/src/utils/focusTrapComponent.ts

```diff
@@ -58,9 +71,10 @@ export function connectFocusTrap(component: FocusTrapComponent, options?: Connec
   }
 
   const focusTrapOptions: FocusTrapOptions = {
-    clickOutsideDeactivates: true,
-    escapeDeactivates: false,
+    clickOutsideDeactivates: component.clickOutsideDeactivates ?? true,
+    escapeDeactivates: !component.escapeDisabled ?? true,
```

Comment:
This can be simplified to `!component.escapeDisabled`.

The `!` operator will cast `component.escapeDisabled` to a boolean, so the nullish coalescing operator won't ever get triggered because it only falls back to the right-hand side value only if the left-hand side is `null` or `undefined`.